ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml

```diff
@@ -76,43 +103,270 @@ StatusDialog {
+                // from the model
+                for (let i = 0; i < model.count; i++) {
+                    const item = model.get(i)
+                    prefixStr += item.shortName + ":"
+                    // Remove all added prefixes from initial array
+                    prefixes = prefixes.filter(e => e !== item.shortName)
+                }
+
+                const unknownPrefixes = getUnknownPrefixes(prefixes)
+                if (unknownPrefixes.length > 0) {
+                    prefixStr += unknownPrefixes.join(":") + ":"
+                }
+
+                return prefixStr
             }
-            readOnly: root.edit || root.addAddress
-            wrongInputValidationError: qsTr("Please enter a valid ENS name OR Ethereum Address")
-            ownAddressError: qsTr("Can't add yourself as a saved address")
         }
+
+        StatusNetworkSelector {
+            id: networkSelector
+
+            title: "Network preference"
+            enabled: addressInput.valid && !d.addressInputIsENS
+            defaultItemText: "Add networks"
+            defaultItemImageSource: "add"
+            rightButtonVisible: true
+
+            property bool modelUpdateBlocked: false
+
+            function blockModelUpdate(value) {
+                modelUpdateBlocked = value
+            }
+
+            itemsModel: SortFilterProxyModel {
+                sourceModel: allNetworksModelCopy
+                filters: ValueFilter {
+                    roleName: "isEnabled"
+                    value: true
+                }
+
+                onCountChanged: {
+                    if (!networkSelector.modelUpdateBlocked) {
+                        // Initially source model is empty, filter proxy is also empty, but does
+                        // extra work and mistakenly overwrites root.chainShortNames property
+                        if (sourceModel.count != 0) {
+                            const prefixAndAddress = Utils.splitToChainPrefixAndAddress(addressInput.plainText)
+                            const syncedPrefix = addressInput.syncChainPrefixWithModel(prefixAndAddress.prefix, this)
+                            root.chainShortNames = syncedPrefix
+                            addressInput.setPlainText(syncedPrefix + prefixAndAddress.address)
+                        }
+                    }
+                }
+            }
+
+            addButton.highlighted: networkSelectPopup.visible
+            addButton.onClicked: {
+                networkSelectPopup.openAtPosition(addButton.x, networkSelector.y + addButton.height + Style.current.xlPadding)
+            }
+
+            onItemClicked: {
+                // Append first item
+                if (index === 0 && defaultItem.visible)
+                    networkSelectPopup.openAtPosition(defaultItem.x, networkSelector.y + defaultItem.height + Style.current.xlPadding)
+            }
+
+            onItemRightButtonClicked: {
```

Comment:
What do you think of using functions with arguments instead `onItemRightButtonClicked: function(item, index, mouse) {`?
Won't this improve clarity and better support in case of parameter change?